src/app/clusters/ota-requestor/GenericOTARequestorDriver.h

```diff
@@ -82,6 +82,9 @@ class GenericOTARequestorDriver : public OTARequestorDriver
     OTAImageProcessorInterface * mImageProcessor = nullptr;
     uint32_t mOtaStartDelaySec                   = 0;
     uint32_t mPeriodicQueryTimeInterval = (24 * 60 * 60); // Timeout for querying providers on the default OTA provider list
+    // Maximum number of times to retry a BUSY OTA provider before moving to the next available one
+    static constexpr uint8_t kMaxBusyProviderRetryCount = 10;
```

Comment:
Ten retries seems excessive. I would have done three.

Reply:
As mentioned this morning, we should apply this retry to `UpdateNotFoundReason::kNotAvailable` as well. Perhaps this value can be more generically named to something like `kMaxProviderRetryCount`